src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -607,22 +607,7 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
     MPIDI_Global.rma_iov_limit = MIN(prov_use->tx_attr->rma_iov_limit, MPIDI_OFI_IOV_MAX);
     MPIDI_Global.max_mr_key_size = prov_use->domain_attr->mr_key_size;
 
-    if (MPIDI_Global.max_mr_key_size >= 8) {
-        MPIDI_Global.max_rma_key_bits = MPIDI_OFI_MAX_KEY_BITS_64;
-        MPIDI_Global.max_huge_rmas = MPIDI_OFI_MAX_HUGE_RMAS_64;
-        MPIDI_Global.context_shift = MPIDI_OFI_CONTEXT_SHIFT_64;
-        MPIDI_Global.rma_key_type_bits = MPIDI_OFI_MAX_KEY_TYPE_BITS_64;
-    } else if (MPIDI_Global.max_mr_key_size >= 4) {
-        MPIDI_Global.max_rma_key_bits = MPIDI_OFI_MAX_KEY_BITS_32;
-        MPIDI_Global.max_huge_rmas = MPIDI_OFI_MAX_HUGE_RMAS_32;
-        MPIDI_Global.context_shift = MPIDI_OFI_CONTEXT_SHIFT_32;
-        MPIDI_Global.rma_key_type_bits = MPIDI_OFI_MAX_KEY_TYPE_BITS_32;
-    } else if (MPIDI_Global.max_mr_key_size >= 2) {
-        MPIDI_Global.max_rma_key_bits = MPIDI_OFI_MAX_KEY_BITS_16;
-        MPIDI_Global.max_huge_rmas = MPIDI_OFI_MAX_HUGE_RMAS_16;
-        MPIDI_Global.context_shift = MPIDI_OFI_CONTEXT_SHIFT_16;
-        MPIDI_Global.rma_key_type_bits = MPIDI_OFI_MAX_KEY_TYPE_BITS_16;
-    } else {
+    if (MPIDI_Global.max_mr_key_size < 2) {
         MPIR_ERR_SETFATALANDJUMP4(mpi_errno,
```

Comment:
Can we briefly explain why error should be reported here ?

Reply:
The original design basically requires the provider to support at 16-bit keys (`max_mr_key_size >= 2 (bytes)`.  Since we use the id space to allocate window id and MR keys for LMT protocol, we want to have a sufficiently large key space. I think this is also related to the Pavan's comment regarding the required OFI feature which we should clarify.

Reply:
Now that we no longer split the key space, I'd think we can remove the requirement for key size.

Reply:
Agreed. We shouldn't put arbitrary restrictions on it.

Reply:
Fixed in the latest revision